trie/patriciaMerkleTrie.go

```diff
@@ -88,19 +90,25 @@ func NewTrie(
 		chanClose:               make(chan struct{}),
 		enableEpochsHandler:     enableEpochsHandler,
 		trieNodeVersionVerifier: tnvv,
+		batchManager:            trieBatchManager.NewTrieBatchManager(),
 	}, nil
 }
 
 // Get starts at the root and searches for the given key.
 // If the key is present in the tree, it returns the corresponding value
 func (tr *patriciaMerkleTrie) Get(key []byte) ([]byte, uint32, error) {
+	hexKey := keyBytesToHex(key)
+	val, found := tr.batchManager.Get(hexKey)
```

Comment:
We use hex keys to update the trie. If I do not save the hexKey in the batch, when we update the batched changes to the trie, we need to go through all the changes and then compute the hex key for each change. So this saves us the time it would take to iterate the batched keys in order to convert them.

Reply:
can be left as it is but you are traversing the map & keys from the batch anyway. Could have been done there, the process time should have been almost the same (maybe the trie.Get function could have called it twice). Will see the memory footprint, if we are going to see abnormal figures. we will need to drop the usage of storing hexed keys

Reply:
Added TODO to keep an eye for the memory usage